frontend/eda/event-streams/EventStreamPage/EventStreamPage.tsx

```diff
@@ -0,0 +1,164 @@
+    }
+  });
+
+  const restartEventStream = useRestartEventStreams((restarted) => {
+    if (restarted.length > 0) {
+      refresh();
+    }
+  });
+
+  const deleteEventStreams = useDeleteEventStreams((deleted) => {
+    if (deleted.length > 0) {
+      pageNavigate(EdaRoute.EventStreams);
+    }
+  });
+
+  const enableEventStream: (eventStream: EdaEventStream) => Promise<void> = useCallback(
+    async (eventStream) => {
+      const alert: AlertProps = {
+        variant: 'success',
+        title: `${eventStream.name} ${t('enabled')}.`,
+        timeout: 5000,
+      };
+      await postRequest(edaAPI`/event-streams/${eventStream.id.toString()}/enable/`, undefined)
+        .then(() => alertToaster.addAlert(alert))
+        .catch(() => {
+          alertToaster.addAlert({
+            variant: 'danger',
+            title: `${t('Failed to enable')} ${eventStream.name}`,
+            timeout: 5000,
+          });
+        });
+      refresh();
+    },
+    [alertToaster, refresh, t]
+  );
+
+  const isActionTab = location.href.includes(
+    getPageUrl(EdaRoute.EventStreamDetails, { params: { id: eventStream?.id } })
+  );
+
+  const itemActions = useMemo<IPageAction<EdaEventStream>[]>(() => {
+    const actions: IPageAction<EdaEventStream>[] = isActionTab
+      ? [
+          {
+            type: PageActionType.Switch,
+            selection: PageActionSelection.Single,
+            ariaLabel: (isEnabled) =>
+              isEnabled ? t('Click to disable instance') : t('Click to enable instance'),
+            isPinned: true,
+            label: t('Event stream enabled'),
+            labelOff: t('Event stream disabled'),
+            onToggle: (eventStream: EdaEventStream, activate: boolean) => {
+              if (activate) void enableEventStream(eventStream);
+              else void disableEventStream([eventStream]);
+            },
+            isSwitchOn: (eventStream: EdaEventStream) => eventStream.is_enabled ?? false,
+            isDisabled: (eventStream: EdaEventStream) =>
+              eventStream.status === Status906Enum.Stopping
+                ? t('Cannot change the event stream status while stopping')
+                : undefined,
+          },
+          {
+            type: PageActionType.Button,
+            selection: PageActionSelection.Single,
+            icon: RedoIcon,
+            label: t('Restart event stream'),
+            isDanger: false,
+            isHidden: (eventStream: EdaEventStream) => !eventStream.is_enabled,
+            onClick: (eventStream: EdaEventStream) => restartEventStream([eventStream]),
+          },
+          {
+            type: PageActionType.Seperator,
+          },
+          {
+            type: PageActionType.Button,
+            selection: PageActionSelection.Single,
+            icon: TrashIcon,
+            label: t('Delete event stream'),
+            onClick: (eventStream: EdaEventStream) => deleteEventStreams([eventStream]),
+            isDanger: true,
+          },
```

Comment:
can we have data-cys added to the above elements here?

Reply:
data-cy already exists - provided by components